File: quasar/Quasar.DataTransfer/Usuarios/Requests/UsuarioCadastroRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Quasar.DataTransfer.Usuarios.Requests
{
    public class UsuarioCadastroRequest
    {
        public string Email { get; set; }
        public string Senha { get; set; }
        [Compare(nameof(Senha), ErrorMessage = "As senhas devem ser iguais")]
        public string SenhaConfirmacao { get; set; }
        public ClienteInserirRequest Cliente { get; set; }
    }
}

File: quasar-front/src/app/usuario/models/request/usuarioCadastroRequest.ts
import { ClienteInserirRequest } from './clienteInserirRequest';
export class UsuarioCadastroRequest {
  public email: string;
  public senha: string;
  public senhaConfirmacao: string;
  public cliente: ClienteInserirRequest

  constructor(params: Partial<UsuarioCadastroRequest>) {
    this.email = params.email || '';
    this.senha = params.senha || '';
    this.senhaConfirmacao = params.senhaConfirmacao || '';
    this.cliente = params.cliente || new ClienteInserirRequest({});
  }
}
